### PR TITLE
Support Iceberg partition identity transform

### PR DESCRIPTION
**Pull request by regadas**
Fixes: #33331

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by regadas** [comment]
retest this please

**Comment by regadas** [comment]
assign set of reviewers

**Comment by regadas** [comment]
@ahmedabu98 can you take a look at this one as well?

File: sdks/java/io/iceberg/build.gradle [hunk 1]
```diff
@@ -45,6 +45,7 @@ dependencies {
     implementation library.java.vendored_guava_32_1_2_jre
     implementation project(path: ":sdks:java:core", configuration: "shadow")
     implementation project(path: ":model:pipeline", configuration: "shadow")
+    implementation library.java.avro
```
Comment (ahmedabu98) [inline sdks/java/io/iceberg/build.gradle:48]: I may have missed it, but did we need this dependency for this PR?

**Review by ahmedabu98 — APPROVED** [review]
LGTM, just one question about the avro dep

**Comment by ahmedabu98** [comment]
waiting on author

File: sdks/java/io/iceberg/build.gradle (hunk 1 above)
Comment (regadas) [reply sdks/java/io/iceberg/build.gradle]: yeah, it's not needed indeed. Staged incorrectly.

**Comment by regadas** [comment]
@ahmedabu98 addressed the comment and rebased it.

File: sdks/java/io/iceberg/build.gradle (hunk 1 above)
Comment (regadas) [reply sdks/java/io/iceberg/build.gradle]: ahah! it seems that it's actually needed?

```
Execution failed for task ':sdks:java:io:iceberg:analyzeClassesDependencies'.
> Dependency analysis found issues.
  usedUndeclaredArtifacts
   - org.apache.avro:avro:1.11.3@jar
```
when I double checked locally I wasn't getting this...

Comment (ahmedabu98) [reply sdks/java/io/iceberg/build.gradle:48]: hmm that's interesting, I checked out the PR and ran `./gradlew :sdks:java:io:iceberg:analyzeClassesDependencies` and got the same error. I don't see any lines in this change that introduce Avro library.

Anyways, I think there's no harm in including it. will merge this

**Pull request merged**